packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/Task/Task.js

```diff
@@ -1,19 +1,33 @@
+import asyncStates from '@zooniverse/async-states'
 import { Box, Paragraph } from 'grommet'
-import { bool, func, shape } from 'prop-types'
-import React, { useEffect, useState } from 'react'
+import { MobXProviderContext, observer } from 'mobx-react'
+import { func, shape, string } from 'prop-types'
+import React, { useContext } from 'react'
 
 import taskRegistry from '@plugins/tasks'
 
-export default function Task (props) {
-  const { classification, disabled, task } = props
-  const [ annotation, setAnnotation ] = useState()
-  const { TaskComponent } = taskRegistry.get(task.type)
+function useStores(task, stores) {
+  const { classifierStore } = stores || useContext(MobXProviderContext)
 
-  useEffect(function onMount() {
-    const annotation = classification.addAnnotation(task)
-    task.setAnnotation(annotation)
-    setAnnotation(annotation)
-  }, [])
+  const {
+    classifications,
+    subjectViewer
+  } = classifierStore
+  const { loadingState: subjectReadyState } = subjectViewer
+  const classification = classifications.active
+  const disabled = subjectReadyState !== asyncStates.success
+  const annotation = classification.addAnnotation(task)
+  task.setAnnotation(annotation)
+  return {
+    annotation,
+    disabled
+  }
+}
+
+function Task (props) {
+  const { store, task } = props
+  const { annotation, disabled } = useStores(task, store)
```

Comment:
Non-blocking suggestion: I think we can still have a default here for disabled by:
```
const { annotation, disabled = false } = useStores(task, store)
```

Reply:
Looking through `useStores`, `disabled` can’t be undefined.